### PR TITLE
[docs] eas update: better navigation titles

### PR DESCRIPTION
**Pull request by quinlanj**
# Why

Improve the readability of the navigation sidebar for EAS Update, similar to EAS Build

# How

- Most articles under the `EAS Update` section should not have `EAS Update` in the article title because it is already placed in the relevant category. EAS Build does this, and it makes it easier to navigate the docs.
- Avoid long titles. For example `How to optimize assets with EAS Update` can be shortened to `Optimizing Assets`

# UX
Before: 
Screenshot 2023-06-19 at 5 02 14 PM

After:
Screenshot 2023-06-19 at 5 02 37 PM

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/eas-update/introduction.mdx [hunk 1]
```diff
@@ -30,7 +30,7 @@ Still using Classic Updates? We moved those docs to [the archive](/archive/class
 />
 
 <BoxLink
-  title="Developing with EAS Update"
+  title="Developing Faster"
```
Comment (kbrandwijk) [inline docs/pages/eas-update/introduction.mdx:33]:
```suggestion
  title="Developing faster"
```

File: docs/pages/eas-update/eas-update-with-local-build.mdx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-title: EAS Update with a local build
+title: Building Locally
```
Comment (kbrandwijk) [inline docs/pages/eas-update/eas-update-with-local-build.mdx:2]:
```suggestion
title: Building locally
```

File: docs/pages/eas-update/developing-with-eas-update.mdx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Developing with EAS Update
+title: Developing Faster
```
Comment (kbrandwijk) [inline docs/pages/eas-update/developing-with-eas-update.mdx:2]:
```suggestion
title: Developing faster
```

**Review by kbrandwijk — COMMENTED** [review]
Left a few suggestions for consistency with other titles throughout this and other sections.

**Comment by Simek** [comment]
AFAIK we added the "EAS Update" to the page titles on purpose, to avoid duplication of the search results and to keep them clear and separated, since, for example, in the past "Debugging" query was returning few pages titled like that, and it was hard for the users to discern which part of the docs/which product etc. given page is devoted to.

**Comment by kbrandwijk** [comment]
> AFAIK we added the "EAS Update" to the page titles on purpose, to avoid duplication of the search results and to keep them clear and separated, since, for example, in the past "Debugging" query was returning few pages titled like that, and it was hard for the users to discern which part of the docs/which product etc. given page is devoted to.

Then these are probably best implemented as `sidebar_title` overrides, leaving the page title as-is for search purposes. wdyt?

**Comment by amandeepmittal** [comment]
We can also change the usage of gerunds to use short-form verbs. For example, "Optimizing assets" to "Optimize assets". This a pattern we've started following in the "Home" section of our docs: https://docs.expo.dev/

**Comment by quinlanj** [comment]
Thanks for the feedback, everyone! I've updated the PR feedback with the following:
- used `sidebar_title`
- preserve `EAS Update` in the page titles, except for CodePush and Classic Updates migration docs, which shouldn't have duplicates since they are only relevant to EAS Update
- modified the page titles and sidebar titles to use short-form verbs (ie) "Optimizing assets" to "Optimize assets"

**Review by kbrandwijk — APPROVED** [review]
Thanks for the fixes! :shipit:

**Pull request merged**